src/lib/components/Swap/Toolbar/Caption.tsx

```diff
@@ -25,21 +26,37 @@ function Caption({ icon: Icon = AlertTriangle, caption }: CaptionProps) {
   )
 }
 
+export function WrapCurrency({ loading, wrapType }: { loading: boolean; wrapType: WrapType.UNWRAP | WrapType.WRAP }) {
+  const WrapText = useCallback(() => {
+    if (wrapType === WrapType.WRAP) {
```

Comment:
I prefer to make this explicit, so that you check against both types. A good way to do this and have tsc ensure completeness is to use switch:
```
switch (wrapType) {
  case WrapType.WRAP:
    ...
  case WrapType.UNWRAP:
    ...
}
```

I prefer to do it this way because WrapType is an enum, not a boolean, so there's not a guarantee (in code) that there are only two cases.

Reply:
Just realized there _is_ a third type to WrapType, but you've already excluded it in your typing on L29.